### ISSUE TITLE
Make pdf-viewer have custom dark background like pdf-tools?

### ISSUE THREAD

**Issue by taquangtrung** [emacs-eaf/eaf-pdf-viewer#15]
Hi,

**Is your feature request related to a problem? Please describe.**

Currently, EAF pdf-viewer can support only the original background color (often white) or inverted background color (white -> black) of a document.

I cannot set a custom background color (In pdf-tools, I can do so by setting the background color in mid-night mode) 

**Describe the solution you'd like**

It would be nice if EAF pdf-viewer can support the custom background color of documents (so that this background color can match with the background theme of Emacs)

**Describe alternatives you've considered**

I looked into this code of `buffer.py` and see that the inverted color is controlled at the pixel level by the function call `pixmap.invertIRect(pixmap.irect)` to PyMuPDF.

```python
    def with_invert(self, invert, exclude_image=True):
        ...
        def fn(page, pixmap, scale):
            pixmap.invertIRect(pixmap.irect)
```

I haven't figured out how to make PyMuPDF only render text in inverted mode and set the background according to users' preferences.

Is this possible to implement such a feature in EAF pdf-viewer?

**Comment by taquangtrung** [comment]
Oh, but I looked at this screenshot  and see that the background color is not black:

https://github.com/emacs-eaf/eaf-pdf-viewer/blob/master/latex_screenshot.gif

Can you advise how to do it? Or that's the default background color of the PDF file?

**Comment by luhuaei** [comment]
> Oh, but I looked at this screenshot and see that the background color is not black:
> 
> https://github.com/emacs-eaf/eaf-pdf-viewer/blob/master/latex_screenshot.gif
> 
> Can you advise how to do it? Or that's the default background color of the PDF file?

The gif background color is "follow"  emacs current theme color.

```elisp
(defcustom eaf-pdf-dark-mode "follow"
  "Whether to enable inverted color rendering when starting the pdf-viewer app.

Possible values are
   - \"follow\" Follow the background color of the theme of user.
   - \"force\" Force inverted color rendering on start-up.
   - \"ignore\" Don't do inverted rendering."
  :type '(choice (string :tag "Force inverted color rendering." "force")
                 (string :tag "Follow the background color of user's theme." "follow")
                 (other :tag "Do normal rendering." "ignore"))
  :group 'eaf-pdf-viewer)
```

**Comment by luhuaei** [comment]
> I haven't figured out how to make PyMuPDF only render text in inverted mode and set the background according to users' preferences.

You can press `i` to `toggle_inverted_mode`. It has follow mode:
```
- normal
- invert mode(only text, the image not effect)
- invert mode(text, image both inverted)
```

**Comment by taquangtrung** [comment]
@luhuaei Thank you for your suggestion!  I tested again and realized that the PDF that I used was rendered by LaTeX Beamer, where the background somehow was set explicitly to the white color.

I used the following code to clear all the settings of Beamer's background canvas, then the background color of the generated PDF file can be changed correctly by EAF pdf-viewer.

```
\setbeamertemplate{background canvas}{}
```
